build.rs

```diff
@@ -162,6 +179,21 @@ fn main() {
         .expect("Cargo-provided environment variables should always exist!");
     let build_desc = tracker.get_env_var("VECTOR_BUILD_DESC");
 
+    // Get the git short hash of the HEAD.
+    // Note that if Vector is compiled within a container, proper git permissions must be set for
+    // the repo directory.
+    // In CI build workflows this will have been pre-configured by running the command
+    // "git config --global --add safe.directory /git/vectordotdev/vector", from the vdev package
+    // subcommands.
+    let git_short_hash = git_short_hash()
+        .map_err(|e| -> String {
+            format!(
+                "Unable to determine git short hash from rev-parse command: {}",
+                e
+            )
+        })
+        .expect("git hash detection failed");
```

Comment:
I think this `expect` will actually swallow the underlying error. Maybe the `map_err` should `eprintln!()`?